### PR TITLE
Add unit tests for com.intuit.karate.core.Tag

### PR DESCRIPTION
**Pull request by petersear**
### Description

Thanks for contributing this Pull Request. Make sure that you send in this Pull Request to the `develop` branch of this repository, add a brief description, and tag the relevant issue(s) and PR(s) below.

- Relevant Issues : #726 #812
- Relevant PRs : (optional)
- Type of change :
  - [ ] New feature
  - [ ] Bug fix for existing feature
  - [ ] Code quality improvement
  - [x] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

I've analysed your codebase and noticed that com.intuit.karate.core.Tag is not fully tested.
I've written some tests with the help of [Diffblue Cover](https://www.diffblue.com/opensource) for the following methods in this class:

- toString()
- equals()

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be happy to look at other classes that you consider important

### CODE REVIEW

**Review by JohnLBergqvist — DISMISSED** [review]

File: karate-core/src/test/java/com/intuit/karate/StringUtilsTest.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * The MIT License
  *
- * Copyright 2017 Intuit Inc.
+ * Copyright 2017-2019 Intuit Inc.
```
Comment (ptrthomas) [inline karate-core/src/test/java/com/intuit/karate/StringUtilsTest.java]: this change is not needed

File: karate-core/src/test/java/com/intuit/karate/FileUtilsTest.java [hunk 2]
```diff
@@ -24,6 +47,117 @@
...
+    public void testIsClassPath() {
+        assertFalse(FileUtils.isClassPath("foo/bar/baz"));
+
+        assertTrue(FileUtils.isClassPath("classpath:foo/bar/baz"));
+    }
+
+    @Test
+    public void testIsFilePath() {
+        assertFalse(FileUtils.isFilePath("foo/bar/baz"));
+
```
Comment (ptrthomas) [inline karate-core/src/test/java/com/intuit/karate/FileUtilsTest.java]: I prefer not having these extra line feeds for unit tests

File: karate-core/src/test/java/com/intuit/karate/core/TagTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    assertEquals("@foobar,baz", new Tag(5, "@foobar,baz").toString());
+  }
+
+  @Test
+  public void testHashcode() {
+    assertEquals(894422763, tag.hashCode());
+  }
+  @Test
+  public void testEquals() {
+    assertTrue(tag.equals(tag));
```
Comment (Braavos96) [inline karate-core/src/test/java/com/intuit/karate/core/TagTest.java]: Add blank line between asserts.
And I like what your doing there :)

File: karate-core/src/test/java/com/intuit/karate/core/TagTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  public void testToString() {
+    assertEquals("@foo=bar,baz", new Tag(5, "@foo=bar,baz").toString());
+    assertEquals("@foo=", new Tag(5, "@foo=").toString());
+    assertEquals("@foobar,baz", new Tag(5, "@foobar,baz").toString());
+  }
+
+  @Test
+  public void testHashcode() {
+    assertEquals(894422763, tag.hashCode());
+  }
```
Comment (Braavos6) [inline karate-core/src/test/java/com/intuit/karate/core/TagTest.java:68]: add blank line

**Review by EricHetti — APPROVED** [review]

File: karate-core/src/test/java/com/intuit/karate/StringUtilsTest.java (hunk 1 above)
Comment (petersear) [reply karate-core/src/test/java/com/intuit/karate/StringUtilsTest.java]: OK, reverted.

File: karate-core/src/test/java/com/intuit/karate/FileUtilsTest.java (hunk 2 above)
Comment (petersear) [reply karate-core/src/test/java/com/intuit/karate/FileUtilsTest.java]: Removed.

**Comment by petersear** [comment]
Thanks @ptrthomas I have made those changes and will now submit the PR to the parent project and close this one (which is against the fork)

**Pull request closed without merging**